io-engine/src/logger.rs

```diff
@@ -416,6 +461,44 @@ impl LogFormat {
         writeln!(writer)
     }
 
+    /// Formats an event in json mode to stdout.
+    fn json_style<S, N>(
+        &self,
+        _context: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> std::fmt::Result
+    where
+        S: tracing_core::subscriber::Subscriber + for<'s> LookupSpan<'s>,
+        N: for<'w> FormatFields<'w> + 'static,
+    {
+        let normalized = event.normalized_metadata();
+        let meta = normalized.as_ref().unwrap_or_else(|| event.metadata());
+        let fmt = FormatLevel::new(meta.level(), self.ansi);
+        let now = chrono::Local::now();
+
+        let mut output_string = String::new();
+        let mut visitor = StringVisitor::new(&mut output_string);
+        event.record(&mut visitor);
+        let output = visitor.string;
+        let output_split: Vec<&str> = output.split('=').collect();
+        let k = output_split.first().unwrap().trim().replace(';', "");
```

Comment:
what is the logic here, we're collecting the first key and value of what?

Reply:
This is the logger example if we dont modify msg which comes from Event object.

"msg":"message = Subsystem start completed: Ok; self = NvmfSubsystem { id: 0, subtype: \"Discovery\", subnqn: \"nqn.2014-08.org.nvmexpress.discovery\", sn: \"00000000000000000000\", mn: \"Mayastor NVMe controller\", allow_any_host: 1, ana_reporting: 0, listeners: Some([Transport ID { trtype: 3, trstring: \"TCP\", traddr: \"127.0.0.1\", trsvcid: \"8420\" }]) }; "

So we split at '=' and then trim (remove whitespace) to make json same as control plane logging.

Reply:
Is there no way of parsing the fields directly rather than generating the string and then parsing the string again?

Reply:
Ill check

Reply:
We're not capturing the tracing fields here, only the message

Reply:
@tiagolobocastro , As discussed we will merge this as is. Then try to use better tracing package in this repo in future.